File: ga/gpt3_5_result_2024_6/result/1968C_cpp_gpt/33_1968C.cpp
use std::io::{self, BufRead};

const N: usize = 500;
const X: usize = 500;
const A: i32 = 1000000000;

fn proc(n: usize, x: Vec<i32>) -> Vec<i32> {
    let mut a: Vec<i32> = vec![0; n];
    a[0] = x[1] + 1;

    for i in 1..n {
        let mut j = 0;
        while j <= A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i] || a[i] != a[i - 1]) {
                break;
            }
            j += 1;
        }
    }

    a
}

fn main() {
    let stdin = io::stdin();
    let mut input = stdin.lock().lines().map(|line| line.unwrap());

    let t: usize = input.next().unwrap().trim().parse().unwrap();
    for _ in 0..t {
        let n: usize = input.next().unwrap().trim().parse().unwrap();
        let x: Vec<i32> = input.next().unwrap().split_whitespace()
            .map(|x| x.parse().unwrap()).collect();

        let result = proc(n, x);
        println!("{}", result.iter().map(|x| x.to_string()).collect::<Vec<String>>().join(" "));
    }
}

File: ga/glm4_result_2024_6/result/1968C_glm/33_1968C.rs
const N: usize = 500;
const X: usize = 500;
const A: i32 = 1_000_000_000;

fn proc() {
    let mut n = 0;
    let mut x = [0; N];
    let mut a = [0; N];

    // Read n and x
    std::io::stdin().read_line(&mut String::new()).expect("Failed to read line");
    n = std::io::stdin().read::<usize>().expect("Failed to read n");
    for i in 1..n {
        x[i] = std::io::stdin().read::<i32>().expect("Failed to read x[i]");
    }

    a[0] = x[1] + 1;

    for i in 1..n {
        let mut j = 0;
        while j <= A as usize {
            a[i] = x[i] + j as i32 * a[i - 1];
            if i + 1 < n && x[i + 1] < a[i] || i + 1 == n && 1 < a[i] {
                break;
            }
            j += 1;
        }
    }

    // Print a
    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let mut t = 0;
    std::io::stdin().read_line(&mut String::new()).expect("Failed to read line");
    t = std::io::stdin().read::<usize>().expect("Failed to read t");
    while t > 0 {
        proc();
        t -= 1;
    }
}